cpp/daal/src/algorithms/linear_model/linear_model_predict_dense_default_batch_impl.i

```diff
@@ -80,14 +81,67 @@ void PredictKernel<algorithmFPType, defaultDense, cpu>::computeBlockOfResponses(
         DAAL_INT iZero             = 0;
         DAAL_INT numBetasValue     = *numBetas;
         DAAL_INT numResponsesValue = *numResponses;
-        for (DAAL_INT j = 0; j < numResponsesValue; j++)
+        for (size_t j = 0; j < numResponsesValue; ++j)
         {
-            Blas<algorithmFPType, cpu>::xxaxpy(numRows, &one, const_cast<algorithmFPType *>(beta + j * numBetasValue), &iZero, responseBlock + j,
-                                               numResponses);
+            Blas<algorithmFPType, cpu>::xxaxpy(numRows, &one, beta + j * numBetasValue, &iZero, responseBlock + j, numResponses);
         }
     } /* if (findBeta0) */
 } /* void PredictKernel<algorithmFPType, defaultDense, cpu>::computeBlockOfResponses */
 
+template <typename algorithmFPType, CpuType cpu>
+services::Status PredictKernel<algorithmFPType, defaultDense, cpu>::computeBlockOfResponsesSOA(
+    const size_t & startRow, DAAL_INT * numFeatures, DAAL_INT * numRowsInBlock, NumericTable * dataTable, DAAL_INT * numBetas,
+    const algorithmFPType * beta, DAAL_INT * numResponses, algorithmFPType * responseBlock, bool findBeta0, algorithmFPType * const TlsData)
+{
+    SafeStatus safeStat;
+    char trans                    = 'T';
+    char notrans                  = 'N';
+    algorithmFPType one           = 1.0;
+    const DAAL_INT nFeatures      = *numFeatures;
+    const DAAL_INT nRowsInBlock   = *numRowsInBlock;
+    const DAAL_INT nResponses     = *numResponses;
+    const DAAL_INT nRowsInData    = dataTable->getNumberOfRows();
+    const DAAL_INT nBlockSizeCols = static_cast<DAAL_INT>(blockSizeColumns);
+    services::internal::service_memset_seq<algorithmFPType, cpu>(responseBlock, algorithmFPType(0.0), nRowsInBlock * nResponses);
+
+    const size_t numBlocks = nFeatures / blockSizeColumns + !!(nFeatures % blockSizeColumns);
+    for (size_t iBlock = 0; iBlock < numBlocks; ++iBlock)
+    {
+        const size_t startCol         = iBlock * blockSizeColumns;
```

Comment:
naming: start**Col**, blockSize**Columns**, num**Cols**InBlock. please align the terminology you use.